### DISCUSSION TITLE
Is there a way to debug or route a log to a sink only if the send failed?

### DISCUSSION THREAD

**Discussion by doron-cohen** [vectordotdev/vector#19587] [Q&A] [answered]
Hi,

We are seeing some failures to send events to GCP PubSub sink. The error is about the event schema `Invalid data in message: Message failed schema validation.`. There are some details but we would like to see the event itself or at least get the event ID in order to find the source of the malformed events.

Is there a way to dump the event or a specific field to Vector stdout logs when an error to send happens? I would imagine an option to route only failed sends to a specific sinks. Maybe if we could query the result of the send in VRL or something.

Thanks

**Comment by jszwedko** [comment, marked as answer]
Unfortunately not yet. This would be covered by https://github.com/vectordotdev/vector/pull/14708 but that's been on the back burner for a bit.
